opm/input/eclipse/Schedule/Well/NameOrder.cpp

```diff
@@ -116,28 +118,50 @@ GroupOrder GroupOrder::serializationTestObject()
 
 void GroupOrder::add(const std::string& gname)
 {
-    auto iter = std::find(this->m_name_list.begin(),
-                          this->m_name_list.end(), gname);
-    if (iter == this->m_name_list.end()) {
-        this->m_name_list.push_back(gname);
+    auto iter = std::find(this->name_list_.begin(),
+                          this->name_list_.end(), gname);
+    if (iter == this->name_list_.end()) {
+        this->name_list_.push_back(gname);
     }
 }
 
 bool GroupOrder::has(const std::string& gname) const
 {
-    auto iter = std::find(this->m_name_list.begin(),
-                          this->m_name_list.end(), gname);
-    return iter != this->m_name_list.end();
+    return std::find(this->name_list_.begin(), this->name_list_.end(), gname)
+        != this->name_list_.end();
 }
 
-const std::vector<std::string>& GroupOrder::names() const
+std::vector<std::string> GroupOrder::names(const std::string& pattern) const
 {
-    return this->m_name_list;
+    auto gnames = std::vector<std::string>{};
+
+    if (pattern.empty()) {
+        return gnames;
+    }
+
+    // Normal pattern matching
+    if (const auto star_pos = pattern.find('*');
```

Comment:
if  `*` is valid pattern, then I would use a special case for this.

Reply:
> if `*` is valid pattern

It's not supposed to be.  Group name matching is a lot more restricted than well name matching.

> then I would use a special case for this.

I can add a check if you like.